gobblin-runtime/src/main/java/org/apache/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -525,11 +532,19 @@ static boolean isCustomizedProgressReportEnabled(Properties properties) {
         && Boolean.parseBoolean(properties.getProperty(ENABLED_CUSTOMIZED_PROGRESS));
   }
 
-  static boolean isMapperFailureFatalEnabled(Properties properties) {
-    return (
-        properties.containsKey(ConfigurationKeys.MR_JOB_MAPPER_FAILURE_IS_FATAL_KEY)
-        && Boolean.parseBoolean(properties.getProperty(ConfigurationKeys.MR_JOB_MAPPER_FAILURE_IS_FATAL_KEY)))
-        || ConfigurationKeys.DEFAULT_MR_JOB_MAPPER_FAILURE_IS_FATAL;
+  static boolean isBooleanPropEnabled(Properties props, String propKey, Optional<Boolean> optDefault) {
```

Comment:
I may be misunderstanding the purpose of this here, but if the way this is being invoked is through passing an `Optional.of(<static_variable>)`, then wouldn't that just resolve to a default value we can use instead of wrapping it around an optional?

Reply:
for the two configs here, there is always a default... but if there weren't any default, that's what the `Optional.absent()` would represent.